<launch>

<arg name="savedata" default="false"/>
<arg name="launch_PID_controller" default="true"/>
<arg name="bag_prefix" default="20211012_noflying_orientation_test"/>
<arg name="mavros_run"          default="true"/>
<arg name="veh_name" default="VOXL1"/>


	<!-- vim: set ft=xml noet : -->
	<!-- example launch script for PX4 based FCU's -->

	<arg name="fcu_url" default="udp://127.0.0.1:14551@:14551" />
	<arg name="gcs_url" default="" />
	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />
	<arg name="fcu_protocol" default="v2.0" />
	<arg name="respawn_mavros" default="false" />

	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
		<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
		<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
	</include>
	
        <!-- Drone Packages -->

        <node pkg="drone_pkg" type="drone_offboard" name="drone_offboard_node" output="screen">
        </node>
        
        <node pkg="drone_pkg" type="servo_control.py" name="servo_control_node" output="screen">
        </node>
        
        <node if="$(arg launch_PID_controller)" pkg="drone_pkg" type="PID_controller" name="PID_controller_node" output="screen">
        </node>

</launch>
